File: golang-devildyw/simple/L1两数之和.go
package main

func main() {
	twoSum([]int{3, 2, 4}, 6)
}

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		if value, exist := m[target-num]; exist {
			return []int{i, value}
		} else {
			m[num] = i
		}
	}

	return []int{}
}

File: java-devildyw/src/main/java/top/dyw/leetcode/simple/L1两数之和.java
package top.dyw.leetcode.simple;

import java.util.HashMap;

public class L1两数之和 {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i=0; i<nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                return new int[]{map.get(target-nums[i]), i};
            }
            map.put(nums[i], i);
        }
        return new int[]{-1,-1};
    }
}
